app/assets/javascripts/admin/addon/controllers/admin-badges/show.js

```diff
@@ -1,35 +1,61 @@
-import { tracked } from "@glimmer/tracking";
+import { cached, tracked } from "@glimmer/tracking";
 import Controller, { inject as controller } from "@ember/controller";
-import { action } from "@ember/object";
-import { next } from "@ember/runloop";
+import { action, getProperties } from "@ember/object";
 import { service } from "@ember/service";
-import { observes } from "@ember-decorators/object";
+import { ajax } from "discourse/lib/ajax";
 import { popupAjaxError } from "discourse/lib/ajax-error";
-import { bufferedProperty } from "discourse/mixins/buffered-content";
 import getURL from "discourse-common/lib/get-url";
 import I18n from "discourse-i18n";
-
-const IMAGE = "image";
-const ICON = "icon";
-
-// TODO: Stop using Mixin here
-export default class AdminBadgesShowController extends Controller.extend(
-  bufferedProperty("model")
```

Comment:
FormKit now uses its own change set (backed by immer.js) have the same behavior than `bufferedProperty` was offering. On top of this it also support nested paths.

If you are not familiar with `bufferedProperty`, this is a mechanism which allows to store mutations on a proxy object without mutating the original object. You can decide to apply these mutations later.